platform-api/src/test/java/io/datacater/core/stream/StreamInspectMethodsTest.java

```diff
@@ -0,0 +1,110 @@
+package io.datacater.core.stream;
+
+import static io.restassured.RestAssured.given;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import io.quarkus.test.common.http.TestHTTPEndpoint;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import java.io.IOException;
+import java.net.URL;
+import java.util.UUID;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import javax.inject.Inject;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.junit.jupiter.api.*;
+
+@QuarkusTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestHTTPEndpoint(StreamEndpoint.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class StreamInspectMethodsTest {
+
+  @Inject
+  @Channel("testStreamInspectMethods")
+  Emitter<ProducerRecord<String, String>> producer;
+
+  UUID uuid;
+
+  @Test
+  @Order(1)
+  void testDefaultInspect()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    start();
+    Response response =
+        given().pathParam("uuid", uuid.toString()).queryParams("limit", "3").get("/{uuid}/inspect");
+
+    Assertions.assertEquals(200, response.getStatusCode());
+    Assertions.assertTrue(response.body().asString().contains("\"partition\":0"));
+    Assertions.assertFalse(response.body().asString().contains("\"partition\":1"));
+    Assertions.assertFalse(response.body().asString().contains("\"partition\":2"));
+  }
+
+  @Test
+  @Order(2)
+  void testDistributedInspect() {
+    Response response =
+        given()
+            .pathParam("uuid", uuid.toString())
+            .queryParams("limit", "10", "sampleMethod", SampleMethod.UNIFORM)
+            .get("/{uuid}/inspect");
+
+    Assertions.assertEquals(200, response.getStatusCode());
+    Assertions.assertTrue(response.body().asString().contains("\"partition\":0"));
+    Assertions.assertTrue(response.body().asString().contains("\"partition\":1"));
+    Assertions.assertTrue(response.body().asString().contains("\"partition\":2"));
```

Comment:
Do we return 9 elements or do we have one partition over-represented here?

Reply:
We would return 10.
we take the ceiling of 10/3 from each partition then trim the sample size to 10. so we would have:
- Partition 0: 4 records
- Partition 1: 4 records
- Partition 2: 2 records